src/lib/gateway/gateway.ts

```diff
@@ -91,8 +91,9 @@ class Gateway extends GatewayBase {
     type: GatewayRequestType,
     request: GatewayRequest
   ) {
+    const txId = request.txId ?? (await this.getFirstUnusedTxId());
```

Comment:
so this behavior should not be done in production?
If yes, I wonder if there's a way to encode this restriction in the code, instead of just documenting it.
For example, changing this strategy based on whether this is in production or not (some flag?).

Reply:
There are two options:
1. Remove the internal handling of the txId and make it mandatory to send custom txId.
2. Differ the production restriction using the `NODE_ENV` var and then throw exception in case no txId is sent when the value is `production`